### DISCUSSION TITLE
About open source agreement

### DISCUSSION THREAD

**Discussion by yiuf63** [Serial-Studio/Serial-Studio#248] [General]
From the link below, Qt Graphs module is still open source under gpl instead of lgpl. This is actually not obvious.
[Qt Graphs](https://doc.qt.io/qt-6/qtgraphs-index.html)
> Qt Graphs is available under commercial licenses from The Qt Company. In addition, it is available under the GNU General Public License, version 3. See Qt Licensing for further details.

Does this conflict with this project under mit?

**Comment by alex-spataru** [comment]
Thank you for pointing out the licensing issue with the Qt Graphs module, which is under GPLv3. I had assumed it was under LGPL, and I’d prefer not to revert to using QWidgets to render plots and then paint them on the QML interface, as it’s very inefficient.

To address this, I think the best solution is to adopt a dual-license model:

1.	**GPLv3 License:** Serial Studio will remain free and open source for those comfortable with GPLv3 requirements.

2.	**Commercial License:** For companies, schools, or other users who cannot comply with GPLv3, a commercial license will be offered. The app will remain the same but include a fee to cover development costs and the Qt commercial license.

This approach keeps Serial Studio open source while supporting professional users and funding future development. I’d love to hear your input or feedback on this decision—thank you for your understanding and support!

**Reply by yiuf63** [reply]
Qt group may made a mistake, which misguid us.
Serial studio is a great open source project, I remind you to prevent Qt company from coming to you. 
Your plan is superb, as it navigates the project clear of any licensing entanglements and paves the way for prosperous growth.
Thank you for writing this project.
